pypdf/generic/_base.py

```diff
@@ -72,15 +72,12 @@ def hash_bin(self) -> int:
         )
 
     def hash_value_data(self) -> bytes:
-        return ("%s" % self).encode()
+        return (f"{self}").encode()
```

Comment:
Do we really need the brackets here anymore?

Reply:
Good call.